### PR TITLE
BaSP-TG: task - added put method

### PR DESCRIPTION
**Pull request by silveriac**

### CODE REVIEW

File: src/resources/tasks.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const newTask = req.body;
+    let taskExists = tasks.filter(task => task.id == newTask.id);
+    let allTasks = tasks;
+    allTasks.forEach((task, index) =>{
+        if(task.id == newTask.id){
+            return allTasks[index] = newTask;
+        };
+    });
+    if(!taskExists[0]){
+        return res.status(404).json({
```
Comment (LucianaPrat) [inline src/resources/tasks.js:17]: Return would not be needed

File: src/resources/tasks.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    };
+    if(!newTask.taskName){
+        return res.status(400).json({
+            success: false,
+            msg: "Task must have a name. "
+        });
+    };
+    fs.writeFile('./src/data/tasks.json', JSON.stringify(allTasks, null, 2), (err) => {
+        if (err) {
+            console.log(err);
```
Comment (LucianaPrat) [inline src/resources/tasks.js:30]: Remove console.log

File: src/resources/tasks.js (hunk 1 above)
Comment (silveriac) [reply src/resources/tasks.js:17]: Return is needed to break out of the function, if it isn't there you'd get the error about sending responses twice

File: src/resources/tasks.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const express = require('express')
+const tasks = require('../data/tasks.json');
```
Comment (nicolasblasco) [inline src/resources/tasks.js:2]: We should use import for libraries

File: src/resources/tasks.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+const fs = require('fs');
+const router = express.Router();
+router.use(express.urlencoded({extended: false}));
```
Comment (nicolasblasco) [inline src/resources/tasks.js:5]: We should move this to index.js

File: src/resources/tasks.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+router.put('/put/', (req, res) =>{
```
Comment (nicolasblasco) [inline src/resources/tasks.js:7]: The route should be / because the method already specifies the action of updating

File: src/resources/tasks.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        };
+        res.status(200).json({
+            success: true,
+            msg: `Task ${newTask.id} modified successfully`,
+            data: newTask
+        });
+    });
+});
+
+module.exports = router;
```
Comment (nicolasblasco) [inline src/resources/tasks.js:42]: When using import this should change to export default

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**